server/channels/app/notification_email.go

```diff
@@ -13,21 +13,140 @@ import (
 	"github.com/pkg/errors"
 
 	"github.com/mattermost/mattermost/server/public/model"
+	"github.com/mattermost/mattermost/server/public/plugin"
 	"github.com/mattermost/mattermost/server/public/shared/i18n"
 	"github.com/mattermost/mattermost/server/public/shared/mlog"
 	"github.com/mattermost/mattermost/server/public/shared/request"
 	email "github.com/mattermost/mattermost/server/v8/channels/app/email"
 	"github.com/mattermost/mattermost/server/v8/channels/utils"
 )
 
-func (a *App) sendNotificationEmail(c request.CTX, notification *PostNotification, user *model.User, team *model.Team, senderProfileImage []byte) error {
+func (a *App) buildEmailNotification(
+	ctx request.CTX,
+	notification *PostNotification,
+	user *model.User,
+	team *model.Team,
+) *model.EmailNotification {
+	channel := notification.Channel
+	post := notification.Post
+	sender := notification.Sender
+
+	translateFunc := i18n.GetUserTranslations(user.Locale)
+	nameFormat := a.GetNotificationNameFormat(user)
+
+	var useMilitaryTime bool
+	if data, err := a.Srv().Store().Preference().Get(
+		user.Id, model.PreferenceCategoryDisplaySettings,
+		model.PreferenceNameUseMilitaryTime); err != nil {
+		useMilitaryTime = false
```

Comment:
This is hard to read 😅 

Additionally, are we intentionally discarding error? Maybe log it with `DEBUG` level?